### DISCUSSION TITLE
Scroll viewport using mouse wheel

### DISCUSSION THREAD

**Discussion by grota** [paperwm/PaperWM#881] [Q&A] [answered]
Ok this might be a dumb question, but is there a way to use the mouse wheel to scroll the viewport (kinda like the wayland gestures do)?

**Comment by Lythenas** [comment, marked as answer]
Currently no. See also #206

But I think this would be very good to add.

**Reply by grota** [reply]
gotcha, thank you!

**Comment by jtaala** [comment]
Well, you can actually achieve this by bringing up the mini-map view, which then responds to mouse scroll.  E.g. default, hit `super+down` (for example) and while continuing to hold down `super`, you can then use your mouse scroll button to scroll left/right.

I've also set this up with something like https://github.com/sezanzeb/input-remapper.  I have it setup so that when I hold down my side button, it triggers `super+down` and holds `super` down so I can then scroll viewport with my mouse button.

**Comment by jtaala** [comment]
E.g I have a keybind to `alt`+`d` which triggers the minimap selects a window and shows the tiling minimap:
![image](https://github.com/paperwm/PaperWM/assets/30424662/9a3af3de-f890-4265-9952-2db7696199f9)


in input-remapper the shortcut looks like this:
![image](https://github.com/paperwm/PaperWM/assets/30424662/63fb64ac-fdbe-4e9d-8671-42e5ab5ad1ab)

Although, this is kind of a workaround - and an implementation to simply show the minimap (and hence allow mouse-scrolling) would be a bit easier.

**Comment by grota** [comment]
Yeah I am currently using a similar workaround:

I copy-pasted the example from input-remapper so that's why you see `foo`  I still need to give it a proper name 😄.

This is for the keyboard:

![image](https://github.com/paperwm/PaperWM/assets/403432/5f088d99-60a7-46d1-a0b4-6255c2419f35)

And this is for the mouse:

![image](https://github.com/paperwm/PaperWM/assets/403432/ea962728-763f-443b-ae98-d971e5110462)

For wheeldown is simmetric: `if_eq($foo, 1, modify(Super_L, key(l)), wheel(down, 45))`

It's nice because if I hold super and use the wheel then it scrolls between workspaces, while if I hold the Left Alt it scrolls inside the current workspace.
The drawback is that every time I disconnect the mouse I need to restart the `input-remapper` "mouse section".

But yeah it would be awesome to have it native to the extension: right now unless I'm mistaken it is not possible to grab a `<left-alt>+wheel-down` here 
![image](https://github.com/paperwm/PaperWM/assets/403432/aa87407d-4b41-40a7-9e91-ed11138f1863)

Or maybe make the two icons in the tob bar ("window focus mode" and "open window position") capture the scroll events (even if it is probably feature creep)

**Reply by jtaala** [reply]
That's not a bad idea - we could even make the top scrolling on the topbar tile scroll left/right (mouse scrolling on the workspace name on the left of the topbar will scroll through workspaces).

The other idea is add a keybind to toggle minimap (e.g. bring up minimap and then you can scroll).

**Comment by jtaala** [comment]
Hey @grota, see #883 which implements this.  ~~Still need to make it work better if users are using the Gnome pill button (rather than PaperWM's workspace indicator).~~

Check it out and give it a go.

Jay.

**Reply by jtaala** [reply]
```
git clone https://github.com/paperwm/PaperWM.git
cd PaperWM
git checkout implement-206-scroll-windows-mouse
./install.sh
```

Then logout/login.

**Reply by grota** [reply]
I tested and it works for me! It's nice that it is still possible to switch workspace using the wheel over the workspace indicator! thank you a lot! 💯 ❤️

**Comment by jtaala** [comment]
This has now been implemented in PaperWM version 46.13.1+.
